test/datastore_test.py

```diff
@@ -1,61 +1,42 @@
-However, that process is time consuming because the data is large and far away.
-Because of that, we don't pull down new data frequently, which means if the
-datastore management infrastructure breaks, we often don't find out about it
-for a long time. These tests are meant to help avoid that problem by
-continuosly exercising this functionality.
-"""
-
-import logging
-import os
-
-import pudl
+"""Exercise the functionality in the datastore management module."""
+
+import random
+
 import pudl.workspace.datastore as datastore
 
-logger = logging.getLogger(__name__)
-
-
-def test_datastore(pudl_settings_fixture, data_scope):
-    """Download sample data for each available data source."""
-    sources_to_update = ['eia860', 'eia923', 'epaipm']
-    years_by_source = {
-        'eia860': data_scope['eia860_years'],
-        'eia923': data_scope['eia923_years'],
-        'epacems': [],
-        'epaipm': [None, ],
-        'ferc1': [],
-    }
-    # Sadly, FERC & EPA only provide access to their data via FTP, and it's
-    # not possible to use FTP from within the Travis CI environment:
-    if os.getenv('TRAVIS'):
-        states = []
-    else:
-        # Idaho has the least data of any CEMS state.
-        states = data_scope['epacems_states']
-        sources_to_update.extend(['ferc1', 'epacems'])
-        years_by_source['ferc1'] = data_scope['ferc1_years']
-        years_by_source['epacems'] = data_scope['epacems_years']
-
-    datastore.parallel_update(
-        sources=sources_to_update,
-        years_by_source=years_by_source,
-        states=states,
-        data_dir=pudl_settings_fixture['data_dir'],
-    )
-
-    pudl.helpers.verify_input_files(
-        ferc1_years=years_by_source['ferc1'],
-        epacems_years=years_by_source['epacems'],
-        epacems_states=states,
-        # Currently no mechanism for automatically verifying EPA IPM files...
-        pudl_settings=pudl_settings_fixture,
-    )
-    data_dir = pudl_settings_fixture['data_dir']
-    pudl.extract.eia860.Extractor(
-        data_dir).verify_years(years_by_source['eia860'])
-    pudl.extract.eia923.Extractor(
-        data_dir).verify_years(years_by_source['eia923'])
+
+class TestDatastore:
+    """Test the Datastore functions."""
+
+    ds = datastore.Datastore(sandbox=True)
+
+    def test_doi_to_url(self):
+        """Get the DOI url right."""
+        number = random.randint(100000, 999999)
+        fake_doi = "10.5072/zenodo.%d" % number
+
+        assert self.ds.doi_to_url(fake_doi) == \
+            "https://sandbox.zenodo.org/api/deposit/depositions/%d" % number
+
+    def test_all_datapackage_json_available(self):
+        """
+        Ensure that every recorded DOI has a datapackage.json file.
+
+        Integration test!
+        """
+
+        def test_it(client, doi):
+            jsr = client.remote_datapackage_json(doi)
+
+            assert "title" in jsr, "No title for %s" % doi
+            assert "resources" in jsr, "No resources for %s" % doi
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=5664147)